shell/components/xen/xen-state.js

```diff
@@ -61,20 +61,24 @@ export default Base => class extends Base {
       this._validator = this._async(this._validate);
     }
   }
+  _propsAndState() {
```

Comment:
This method and its use below is probably hotly debatable. A case could be made that it improves code maintainability (less verbose, easier to add/remove a param from the fairly long list). But it probably adds some performance overhead to create the array and then unpack it to pass along to the params, and is maybe more obtuse.

I'm happy to remove but was interested to hear your design opinion thoughts. I am myself on the fence but enjoyed the opportunity to imaginatively use `...` while reading the code.

Reply:
I'd rather not make this change because of the downsides you describe.

Reply:
I would probably be ok if you did something like this in _validate:
```
const stateArgs = [this._props, this._state, this._lastProps, this._lastState];
...
this._update(...stateArgs);
```

Reply:
We need to rebuild the args array at line 93, since we change `_lastProps` and `_lastState`. Which is what led me to move it all to a helper method, for DRY.

I'll just revert this part of the change, but the rest of it sounds worth landing if you're still supportive.

Reply:
`didUpdate` should be above the assignments to `last...`. Then we can have the static args array.